lib/src/main/java/graphql/nadel/engine/NadelIncrementalResultSupport.kt

```diff
@@ -23,16 +23,20 @@ import java.util.concurrent.atomic.AtomicBoolean
 import java.util.concurrent.atomic.AtomicInteger
 
 class NadelIncrementalResultSupport internal constructor(
-    private val accumulator: NadelIncrementalResultAccumulator,
+    lazyAccumulator: Lazy<NadelIncrementalResultAccumulator>,
```

Comment:
This is a costly object, we should put it behind lazy because most requests aren't using `@defer`

Either way it should only be instantiated once we actually need it, so that the request can proceed quicker.